### PR TITLE
Fuchsia step 1: add SDK version file and artifact download

### LINKED ISSUE #31102: Integrate Fuchsia core sdk into flutter_tool
Download the SDK so we can make use of it

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Adds a version file to the bin/internal with the current (arbitrary) hash of the fuchsia sdk to download. Adds the ability to run `flutter precache --fuchsia` on non-stable branches.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/31102

## Tests

I added the following tests:
- Updated existing precache test.

*Replace this with a list of the tests that you added as part of this PR. A change in behaviour with no test covering it
will likely get reverted accidentally sooner or later. PRs must include tests for all changed/updated/fixed behaviors. See [Test Coverage].*

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/precache.dart [hunk 1]
```diff
@@ -61,6 +63,9 @@ class PrecacheCommand extends FlutterCommand {
       if (argResults['macos']) {
         requiredArtifacts.add(DevelopmentArtifact.macOS);
       }
+      if (argResults['fuchsia']) {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/commands/precache.dart]: How hard would it be to split off the web, linux, etc. artifacts into an UnstableDevelopmentArtifacts? Then you could do something like.
```
for (DevelopmentArtifact artifact in DevelopmentArtifacts.values) {
  if (argResults[artifact.toString()]) {
    requiredArtifacts.add(artifact);
  }
}
if (!FlutterVersion.instance.isStable) {
  for (UnstableDevelopmentArtifact artifact in UnstableDevelopmentArtifacts.values) {
    if (argResults[artifact.toString()]) {
      requiredArtifacts.add(artifact);
    }
  }
}
```
Or maybe put the unstable development artifacts beyond some dummy marker in DevelopmentArtifacts and then iterate over the indices of DevelopmentArtifacts.values before/after that marker.

**Review by zanderso — APPROVED** [review]
lgtm w/ possible cleanup

File: packages/flutter_tools/lib/src/commands/precache.dart (hunk 1 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/commands/precache.dart]: Updated DevelopmentArtifacts to be a class with a name/unstable marker, ptal?

File: packages/flutter_tools/lib/src/cache.dart [hunk 2]
```diff
@@ -15,28 +15,53 @@ import 'base/os.dart';
...
+  static const DevelopmentArtifact linux = DevelopmentArtifact._('linux', unstable: true);
+
+  /// Artifacts required for Fuchsia.
+  static const DevelopmentArtifact fuchsia = DevelopmentArtifact._('fuchsia', unstable: true);
 
   /// Artifacts required by all developments.
-  universal,
+  static const DevelopmentArtifact universal = DevelopmentArtifact._('universal');
+
+  /// THe vaulues of DevelopmentArtifacts.
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/cache.dart]: THe -> The

File: packages/flutter_tools/lib/src/cache.dart [hunk 3]
```diff
@@ -15,28 +15,53 @@ import 'base/os.dart';
 import 'base/platform.dart';
 import 'globals.dart';
 
-/// A tag for a set of development artifacts that need to be cached.
-enum DevelopmentArtifact {
+class DevelopmentArtifact {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/cache.dart]: Maybe retain a comment here.

**Review by zanderso — APPROVED** [review]
still lgtm w/ a couple more nits.

**Pull request merged**